lib/sles4sap/sdaf_library.pm

```diff
@@ -182,8 +217,152 @@ sub create_sdaf_os_var_file {
     my ($entries) = @_;
     croak 'Expected an ARRAYREF but got: ' . ref $entries if (ref $entries ne 'ARRAY');
 
-    write_sut_file(variable_file, join("\n", @$entries));
-    assert_script_run('source ' . variable_file, quiet => 1);
+    write_sut_file(env_variable_file, join("\n", @$entries));
+    assert_script_run('source ' . env_variable_file, quiet => 1);
+}
+
+=head2 set_os_variable
+
+    set_os_variable($variable_name, $variable_value);
+
+B<$variable_name>: Variable name
+
+B<$variable_value>: Variable value. Empty value is accepted as well.
+
+Adds or replaces existing OS env variable value in env variable file (see function ''). File is sourced afterwards to
+load the value. Croaks with incorrect usage.
+
+B<WARNING>: This is executed via 'assert_script_run' therefore output will be visible in logs
+
+=cut
+
+sub set_os_variable {
+    my ($variable_name, $variable_value) = @_;
+    croak 'Missing mandatory argument "$variable_name"' unless $variable_name;
+
+    my $env_variable_file = env_variable_file();
+    my $variable_present = !script_run("grep 'export $variable_name=' $env_variable_file") ? 1 : 0;    # Switch from RC to true/false
+
+    if ($variable_present) {
+        file_content_replace($env_variable_file, "export $variable_name=.*" => "export $variable_name=\"$variable_value\"");
+    }
+    else {
+        assert_script_run("echo 'export $variable_name=\"$variable_value\"' >> $env_variable_file");
+    }
+
+    # Activate new variable
+    load_os_env_variables();
```

Comment:
so function is not just updating the file but also reloading the environment.
If it is so, function doing 2 things, I suggest to remove `load_os_env_variables` from here and move it in the caller, just after the call to `set_os_variable`

Reply:
Those are two tasks but they belong together. Updating file value itself does not set os variable, nor does reading the file without updating it first.